### PR TITLE
feat: add `contributorsjenkinsio` storage account

### PR DESCRIPTION
**Pull request by lemeurherve**
This PR creates a resource group and a storage account to store the static HTML content generated by https://github.com/jenkins-infra/contributor-spotlight/pull/15 and which will be mounted as volume to be served on contributors.origin.jenkins.io from `publick8s` cluster. (contributors.jenkins.io will be served by the CDN Fastly)
This content doesn't need to be accessible from elsewhere, hence the network protection.

Ref:
- https://github.com/jenkins-infra/helpdesk/issues/3809#issuecomment-1822357561

### CODE REVIEW

**Comment by lemeurherve** [comment]
> Plan: 2 to add, 0 to change, 0 to destroy.

**Comment by lemeurherve** [comment]
New resources for a service not yet in production, self-merging to continue on the subject.

**Comment by lemeurherve** [comment]
Note: hotfixed by removing network rules preventing the creation of the File Share from #524
- https://github.com/jenkins-infra/azure/commit/6aeae448be7ef8622e4dd52db1be49f9ec8103ce

File: contributors.jenkins.io.tf [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  location = var.location
+  tags     = local.default_tags
+}
+
+resource "azurerm_storage_account" "contributorsjenkinsio" {
+  name                      = "contributorsjenkinsio"
+  resource_group_name       = azurerm_resource_group.contributors_jenkins_io.name
+  location                  = azurerm_resource_group.contributors_jenkins_io.location
+  account_tier              = "Standard"
+  account_replication_type  = "GRS"
```
Comment (dduportal) [inline contributors.jenkins.io.tf:12]: What is the reasoning behind using `GRS` here? (vs. `LRS` or `ZRS`) . Could you explain it in the issue (and the eventual cost impact)

**Pull request merged**